src/player.cpp

```diff
@@ -10483,11 +10483,19 @@ bool player::can_wield( const item &it, bool alert ) const
             if( alert ) {
                 add_msg( m_info, _("The %s can't be wielded with only one arm."), it.tname().c_str() );
             }
+            if( worn_with_flag("RESTRICT_HANDS") ) {
+                add_msg( m_info, _("The %s you are wearing hinders the use of both hands."),
+                         it.tname().c_str() );
+            }
         } else {
             if( alert ) {
                 add_msg( m_info, _("You are too weak to wield %s with only one arm."),
                          it.tname().c_str() );
             }
+            if( worn_with_flag("RESTRICT_HANDS") ) {
+                add_msg( m_info, _("The %s you are wearing hinders the use of both hands."),
+                         it.tname().c_str() );
```

Comment:
`it` refers to the item you're trying to wield, not the item that restricts wielding.
For example, if you tried to wield a battle axe while already wearing a briefcase, you'd get a message "The battle axe you are wearing...".

Currently there is no good way to get the first item with a given flag. You can either use a generic message (that doesn't refer to the item that blocks wielding, just "some item") or add a new function that checks all worn items for the flag and returns a reference to the first item it encounters that has it.